frontend/public/components/pod-logs.jsx

```diff
@@ -1,194 +1,78 @@
+  return !_.isEmpty(_.get(container, 'status.terminated'));
 };
 
-const dataHasHTML = (data) => {
-  return _.includes(data, '<html') || _.includes(data, '<HTML');
+// Component to render resource dropdown or just the resource name if only
+// one resource is provided.
+const ContainerDropdown = ({currentContainer, containers, kind, onChange}) => {
+  const currentResourceNameWithIcon = <ResourceName name={currentContainer.name || <LoadingInline />} kind={kind} />;
+
+  if (containers.length > 1) {
+    const dropdownItems = _.map(containers, ({name}) => {
+      return <ResourceName name={name} kind={kind} />;
+    });
+    return <Dropdown className="btn-group" items={dropdownItems} title={currentResourceNameWithIcon} onChange={onChange} />;
+  }
+  return currentResourceNameWithIcon;
 };
 
-export class PodLogs extends SafetyFirst {
+export class PodLogs extends React.Component {
   constructor(props) {
     super(props);
-
-    this._buffer = lineBuffer(1000);
-    this._loadTime = Date.now();
-    this._pendingReload = null;
-
-    this._touchLoadTimeState = _.throttle(this._touchLoadTimeState, 100);
     this._selectContainer = this._selectContainer.bind(this);
-    this._updateLogState = this._updateLogState.bind(this);
-    this._toggleLogState = this._toggleLogState.bind(this);
 
     this.state = {
-      containerNames: [],
-      currentContainer: '',
-      logURL: '',
-      loadTime: 0,
-      logState: 'loading'
+      containers: [],
+      currentContainer: {},
+      eof: false
     };
-    this.state = _.defaults({}, this._initialState(), this.state);
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this._beginStreaming();
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.setState(this._initialState(nextProps.obj));
-  }
-
-  componentWillUnmount() {
-    super.componentWillUnmount();
-    this._endStreaming();
   }
 
-  _initialState(obj = this.props.obj) {
+  static getDerivedStateFromProps(nextProps, prevState) {
     const newState = {};
 
-    const containers = _.get(obj, 'spec.containers', []);
-    newState.containerNames = _.map(containers, 'name');
+    const containers = _.get(nextProps.obj, 'status.containerStatuses', []);
+    newState.containers = _.map(containers, (container) => {
+      return {
+        name: container.name,
+        status: container.state
+      };
+    });
 
-    if (!this.state.currentContainer && newState.containerNames.length > 0) {
-      newState.currentContainer = newState.containerNames[0];
+    if (_.isEmpty(prevState.currentContainer) && newState.containers.length > 0) {
+      newState.currentContainer = newState.containers[0];
     }
 
-    const currentContainer = newState.currentContainer || this.state.currentContainer;
-    newState.logURL = this._logURL(obj, currentContainer);
-
+    const currentContainer = newState.currentContainer || prevState.currentContainer;
+    newState.eof = isContainerTerminated(currentContainer);
     return newState;
```

Comment:
Typically, we'd only want to do this if `newState` differs from the old state, but I doubt Pods are updated often enough to impact performance.